palladian-commons/src/test/java/ws/palladian/helper/StopWatchTest.java

```diff
@@ -0,0 +1,150 @@
+
+    Assert.assertEquals(0L, stopWatch.getStartTime());
+  }
+
+  @PrepareForTest({System.class, StopWatch.class})
+  @Test
+  public void testGetTotalElapsedTime() {
+
+    PowerMockito.mockStatic(System.class);
+
+    final StopWatch stopWatch = new StopWatch();
+    stopWatch.setCountDown(0L);
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(0L).thenReturn(6_231L).thenReturn(95_648L);
+
+    Assert.assertEquals(0L, stopWatch.getTotalElapsedTime());
+    Assert.assertEquals(6_231L, stopWatch.getTotalElapsedTime());
+    Assert.assertEquals(95_648L, stopWatch.getTotalElapsedTime());
+  }
+
+  @Test
+  public void testGetCountDown() {
+
+    final StopWatch stopWatch = new StopWatch();
+
+    stopWatch.setCountDown(0L);
+    Assert.assertEquals(0L, stopWatch.getCountDown());
+
+    stopWatch.setCountDown(5_432L);
+    Assert.assertEquals(5_432L, stopWatch.getCountDown());
+  }
+
+  @PrepareForTest({System.class, StopWatch.class})
+  @Test
+  public void testGetElapsedTime() {
+
+    PowerMockito.mockStatic(System.class);
+
+    final StopWatch stopWatch = new StopWatch();
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(0L).thenReturn(0L).thenReturn(0L).thenReturn(4_758L);
+
+    stopWatch.start();
+    stopWatch.stop();
+
+    Assert.assertEquals(0L, stopWatch.getElapsedTime(false));
+    Assert.assertEquals(0L, stopWatch.getElapsedTime(true));
+
+    stopWatch.start();
+    stopWatch.stop();
+
+    Assert.assertEquals(4_758L, stopWatch.getElapsedTime(false));
+    Assert.assertEquals(4L, stopWatch.getElapsedTime(true));
+  }
+
+  @PrepareForTest({DateHelper.class, StopWatch.class, System.class})
+  @Test
+  public void testGetElapsedTimeStringAndIncrement() {
+
+    PowerMockito.mockStatic(System.class);
+
+    final StopWatch stopWatch = new StopWatch();
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(0L);
+
+    Assert.assertEquals("0ms (+0ms)", stopWatch.getElapsedTimeStringAndIncrement());
+  }
+
+  @PrepareForTest({StopWatch.class, System.class})
+  @Test
+  public void testGetTotalElapsedTimeString() {
+
+    PowerMockito.mockStatic(System.class);
+
+    final StopWatch stopWatch = new StopWatch();
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(0L);
+
+    stopWatch.stop();
+
+    Assert.assertEquals("0ms", stopWatch.toString());
+    Assert.assertEquals("0ms", stopWatch.getTotalElapsedTimeString(false));
+    Assert.assertEquals("0ms", stopWatch.getTotalElapsedTimeString(true));
+
+    stopWatch.setOutputDetail(null);
+
+    Assert.assertEquals("0ms", stopWatch.getTotalElapsedTimeString());
+  }
+
+  @PrepareForTest({DateHelper.class, StopWatch.class, System.class})
+  @Test
+  public void testGetTotalElapsedTimeString2() {
+
+    PowerMockito.mockStatic(System.class);
+
+    final StopWatch stopWatch = new StopWatch();
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(0L);
+
+    Assert.assertEquals("0ms", stopWatch.toString());
```

Comment:
These 3 asserts look identical to those in the method above, testGetTotalElapsedTimeString().

Reply:
I have now condesned them down to two lines but any less than 2 and you will loose code coverage.